hoomd/mpcd/DryingDropletStreaming.h

```diff
@@ -0,0 +1,113 @@
+ * particles in Spherical geometry.
+ *
+ * \tparam Geometry The confining geometry (SphereGeometry).
+ *
+ * The integration scheme is essentially Verlet with specular reflections. The particle is streamed forward over
+ * the time interval. If it moves outside the Geometry, it is placed back on the boundary and its velocity is
+ * updated according to the boundary conditions. Streaming then continues until the timestep is completed.
+ *
+ * To facilitate this, every Geometry must supply three methods:
+ *  1. detectCollision(): Determines when and where a collision occurs. If one does, this method moves the
+ *                        particle back, reflects its velocity, and gives the time still remaining to integrate.
+ *  2. isOutside(): Determines whether a particles lies outside the Geometry.
+ *  3. validateBox(): Checks whether the global simulation box is consistent with the streaming geometry.
+ *
+ */
+
+class PYBIND11_EXPORT DryingDropletStreamingMethod : public mpcd::ConfinedStreamingMethod<mpcd::SphereGeometry>
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param geom Streaming geometry
+         * \param R is the radius of sphere
+         */
+        DryingDropletStreamingMethod(std::shared_ptr<mpcd::SystemData> sysdata,
+                                    unsigned int cur_timestep,
+                                    unsigned int period,
+                                    int phase, std::shared_ptr<::Variant> R, boundary bc)
+        : mpcd::ConfinedStreamingMethod<mpcd::SphereGeometry>(sysdata, cur_timestep, period, phase, std::shared_ptr<mpcd::SphereGeometry>()),
+          m_R(R),m_bc(bc)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+    private:
+        std::shared_ptr<::Variant> m_R; //!Radius of Sphere
+        const boundary m_bc;            //!boundary conditions
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+
+void DryingDropletStreamingMethod<mpcd::SphereGeometry>::stream(unsigned int timestep)
+    {
+    const double start_R = (*m_R)(timestep);
+    const double end_R = (*m_R)(timestep + m_period);
+    const double V = (end_R - start_R)/(m_period * m_mpcd_dt);
+    if (V>0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+    m_validate_geom = m_geom; //validating the geometry
+    m_geom = std::make_shared<const mpcd::SphereGeometry>(end_R, V, m_bc);
+    //stream using ConfinedStreamingMethod
+    ConfinedStreamingMethod<mpcd::SphereGeometry>::stream(timestep);
+    //delete marked particles
+    ArrayHandle<unsigned char> h_bounced(m_bounced, access_location::host, access_mode::overwrite);
+    
```

Comment:
Some comments and whitespace would make this more readable. There are also some small fixes in here, like using `Scalar` not `double` and making sure you `read` the bounced array, not overwrite.
```suggestion
    // compute final radius and average velocity of interface
    const Scalar start_R = (*m_R)(timestep);
    const Scalar end_R = (*m_R)(timestep + m_period);
    const Scalar V = (end_R - start_R)/(m_period * m_mpcd_dt);
    if (V > 0)
        {
        throw std::runtime_error("Droplet radius must decrease.");
        }

    /*
     * Update the geometry, and validate if the initial geometry was not set.
     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
     * is set.
     */
    m_validate_geom = m_geom;
    m_geom = std::make_shared<const mpcd::SphereGeometry>(end_R, V, m_bc);

    // Stream according to base class rules
    ConfinedStreamingMethod<mpcd::SphereGeometry>::stream(timestep);

    // Remove the right amount of bounced particles
    ArrayHandle<unsigned char> h_bounced(m_bounced, access_location::host, access_mode::read);    
```